eno-core/src/main/java/fr/insee/eno/core/reference/EnoCatalog.java

```diff
@@ -124,6 +124,7 @@ private void gatherLabels(EnoQuestionnaire enoQuestionnaire) {
                 .map(EnoTable.class::cast).map(EnoTable::getNoDataCells).forEach(this::gatherLabelsFromNoDataCells);
         // Code lists
         enoQuestionnaire.getCodeLists().stream().map(CodeList::getCodeItems).forEach(this::gatherLabelsFromCodeItems);
+        enoQuestionnaire.getFakeCodeLists().stream().map(CodeList::getCodeItems).forEach(this::gatherLabelsFromCodeItems);
```

Comment:
C'est une meilleure pratique de return plutôt que d'alimenter une var globale

```suggestion
        List<TypeduLabel> codeItemLabes = enoQuestionnaire.getFakeCodeLists().stream().map(CodeList::getCodeItems).map(this::gatherLabelsFromCodeItems).flatmap(Collection::stream).toList();
```
avec un `gatherLabelsFromCodeItems` qui return une liste de codeItemLabel

Tu fais pareil avec `gatherLabelsFromNoDataCells`. et tu peux ajouter le result à celui de `gatherLabelsFromCodeItems` car tu mets tout dans label

et ensuite dans le reste du code tu ajoutes au label le result.

Ca rend les fonctions `gatherLabelsFromCodeItems` et `gatherLabelsFromNoDataCells` testable si jamais y'a besoin.
Et c'est plus clair sur ce qu'elles font car on a le result

Reply:
Oui bien d'accord avec ça

D'ailleurs cette partie de la code base est pas joli joli (enfin c'est une litote 😅 ) : toute la classe "EnoCatalog" a quasi un seul usage c'est celui de ramasser les objets Label qu'on trouve un peu partout pour les traiter ensuite, j'ai dans un coin de la tête de m'en débarrasser pour une classe moins fourre-tout qui dit un peu plus précisément ce qu'elle fait.

Mais oui ça serait déjà ça de mieux avant un refacto un peu plus large !